lib/effects/effects.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/animation.dart';
+import 'package:meta/meta.dart';
+
+import 'dart:math';
+import 'dart:ui';
+
+import '../position.dart';
+import '../components/component.dart';
+
+abstract class PositionComponentEffect {
+  void update(double dt);
+  bool hasFinished();
+  PositionComponent component;
+}
+
+double _direction(double p, double d) => p < d ? -1 : 1;
+double _size(double a, double b) => (a - b).abs();
+
+class ScaleEffect extends PositionComponentEffect {
```

Comment:
should we move these each to their own file? as we add more effects this file my get big (I'm not sure if I have a strong opinion about either option though)